src/SourceBuild/content/eng/pipelines/templates/steps/publish-test-results-pr.yml

```diff
@@ -0,0 +1,207 @@
+
+        for test_file in $updated_test_files; do
+          filename=$(basename "$test_file" | sed 's/Updated//g' | sed 's/\..*$//')
+          files["$filename"]+=" $test_file"
+        done
+
+        for filename in "${!files[@]}"; do
+          updated_test_files="${files[$filename]}"
+
+          if [[ $filename == *"Exclusions"* ]]; then
+            # Exclusion files get combined.
+            # There can only be one exclusion file per key, so we need to check for a file matching the key.
+
+            original_test_file=$(find "$absolute_target_path" -name "${filename}.*")
+            if [ -z "$original_test_file" ]; then
+              CopyNewFilesAndUpdateGit $absolute_target_path $updated_test_files
+            elif [[ "${{ parameters.pipelineId }}" == "$(SOURCE_BUILD_SDK_DIFF_TESTS_PIPELINE_ID)" ]]; then
+              UnionExclusions $original_test_file $updated_test_files
+            elif [[ "${{ parameters.pipelineId }}" == "$(SOURCE_BUILD_LICENSE_SCAN_PIPELINE_ID)" ]]; then
+              IntersectionExclusions $original_test_file $updated_test_files
+            else
+              echo "Unsure of how to combine exclusions files for pipeline type. Exiting."
+              exit 1
+            fi
+          else
+            for updated_test_file in $updated_test_files; do
+              # Non-exclusion files get copied over.
+              # There can be multiple files per key, so we need to check for a file matching each updated file.
+
+              original_test_filename=$(basename $updated_test_file | sed 's/Updated//g')
+              original_test_filepath=$(find "$absolute_target_path" -name "$original_test_filename")
+              if [ -z "$original_test_filepath" ]; then
+                if [[ "${{ parameters.pipelineId }}" == "$(SOURCE_BUILD_LICENSE_SCAN_PIPELINE_ID)" ]]; then
+                  CreateLicenseBaseline $absolute_target_path $updated_test_file
+                else
+                  CopyNewFilesAndUpdateGit $absolute_target_path $updated_test_file
+                fi
+              else
+                cp $updated_test_file $original_test_filepath
+                git add $original_test_filepath
+              fi
+            done
+          fi
+        done
+      }
+
+      function CreatePr() {
+        if [[ -z $(git status --porcelain) ]]; then
+          echo "No changes to commit. Exiting."
+          exit 0
+        fi
+
+        git commit -m "Update test baselines from pipeline ${{ parameters.pipelineId }} build $(Build.BuildId)"
+        git push -u origin "${new_branch_name}"
+
+        readarray -d '/' -t fork_repo_split <<< "${fork_repo}"
+        fork_owner="${fork_repo_split[0]}"
+
+        title="${{ parameters.pullRequestTitle }}"
+
+        echo "TargetRepo: $target_repo"
+        echo "ForkRepo: $fork_repo"
+        echo "Title: $title"
+        echo "Body: $body"
+
+        # create pull request
+        gh pr create \
```

Comment:
Ideally, we'd have logic that would avoid creating duplicate PRs. Let's say a PR gets generated and no one merges it. Then another diff run occurs and generates another PR. Now we've got two PRs to deal with. This also wastes compute resources. So ideally, we'd either no-op if the content matched the existing PR or, if the content had changed, update the existing PR with the new content.

I'd be fine if that logic was done as follow-up work so we can take advantage of this sooner.